app/src/main/java/com/andela/gkuti/mipista/service/LocationDetector.java

```
@@ -0,0 +1,179 @@
+    private double latitude;
+    private Intent intent;
+    private String newLocation = "";
+
+    /**
+     * Constructor for LocationDetector class
+     *
+     * @param context activity context
+     */
+    public LocationDetector(Context context) {
+        this.context = context;
+        intent = new Intent("location");
+    }
+
+    /**
+     * called when connection was successful
+     */
+    @Override
+    public void onConnected(@Nullable Bundle bundle) {
+        if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION)
+                == PackageManager.PERMISSION_GRANTED) {
+            LocationRequest locationRequest = LocationRequest.create();
+            locationRequest.setPriority(LocationRequest.PRIORITY_HIGH_ACCURACY);
+            locationRequest.setInterval(60000);
+            LocationServices.FusedLocationApi.requestLocationUpdates(googleApiClient, locationRequest, this);
+        }
+    }
+
+    /**
+     * called when the connection was suspended
+     */
+    @Override
+    public void onConnectionSuspended(int i) {
+    }
+
+    /**
+     * called when the connection failed
+     */
+    @Override
+    public void onConnectionFailed(@NonNull ConnectionResult connectionResult) {
+    }
+
+    /**
+     * method for setting up a connection
+     */
+    public void connect() {
+        country = " ";
+        state = " ";
+        locality = " ";
+        street = " ";
+        googleApiClient = new GoogleApiClient.Builder(context).addApi(LocationServices.API)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .build();
+        googleApiClient.connect();
+    }
+
+    /**
+     * method for disconnecting the connection
+     */
+    public void disconnect() {
+        try {
+            LocationServices.FusedLocationApi.removeLocationUpdates(
+                    googleApiClient, this);
+        } catch (Exception e) {
```

*Nitpick*: Not sure why you have an empty catch block here. Never fail silently.